library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -167,13 +167,15 @@ public DownloadBatch retrieveBatchFor(DownloadInfo download) {
     }
 
     public void deleteMarkedBatchesFor(Collection<DownloadInfo> downloads) {
-        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, null, SELECT_MARKED_FOR_DELETION, null, null);
+        String[] projection = {Downloads.Impl.Batches._ID};
+        String selection = Batches.COLUMN_DELETED + " = ?";
+        String[] selectionArgs = {MARKED_FOR_DELETION};
```

Comment:
so Andrei originally said do this
`private static final String[] SELECT_DELETED = {Downloads.Impl.Batches._ID};`
`private static final String[] WHERE_MARKED_FOR_DELETION = {"1"};`